NitroxServer/GameLogic/Entities/Spawning/BatchEntitySpawner.cs

```diff
@@ -202,27 +190,99 @@ private IEnumerable<Entity> SpawnEntitiesStaticly(EntitySpawnPoint entitySpawnPo
 
         private IEnumerable<Entity> CreateEntityWithChildren(EntitySpawnPoint entitySpawnPoint, UnityEngine.Vector3 scale, TechType techType, int cellLevel, string classId, DeterministicBatchGenerator deterministicBatchGenerator)
         {
-            Entity spawnedEntity = new Entity(entitySpawnPoint.Position,
-                                              entitySpawnPoint.Rotation,
+            Entity spawnedEntity = new Entity(entitySpawnPoint.LocalPosition,
+                                              entitySpawnPoint.LocalRotation, 
                                               scale,
                                               techType,
                                               cellLevel,
                                               classId,
                                               true,
                                               deterministicBatchGenerator.NextGuid());
+
+            List<Entity> prefabs = GetEntitiesFromClassId(classId, spawnedEntity, deterministicBatchGenerator);
+
             yield return spawnedEntity;
 
             IEntityBootstrapper bootstrapper;
             if (customBootstrappersByTechType.TryGetValue(techType, out bootstrapper))
             {
                 bootstrapper.Prepare(spawnedEntity, deterministicBatchGenerator);
 
-                foreach(Entity childEntity in spawnedEntity.ChildEntities)
+                for (int i = 0; i < spawnedEntity.ChildEntities.Count - prefabs.Count; i++)
                 {
-                    yield return childEntity;
+                    Entity childEntity = spawnedEntity.ChildEntities[i + prefabs.Count];
                 }
             }
         }
 
+        private IEnumerable<Entity> CreateEntityCellRoot(EntitySpawnPoint entitySpawnPoint, DeterministicBatchGenerator deterministicBatchGenerator)
```

Comment:
Do we need to modify the deserialize patch to delete UWE's existing cell roots (we explicitly prevented it from doing that)

Reply:
Might want to I just left em as is.